content/src/service/ServiceImpl.ts

```diff
@@ -63,10 +60,9 @@ export class ServiceImpl implements MetaverseContentService {
     files: DeploymentFiles,
     entityId: string,
     auditInfo: LocalDeploymentAuditInfo,
-    context: DeploymentContext,
-    task?: Database
+    context: DeploymentContext
   ): Promise<DeploymentResult> {
-    const deployedEntity = await this.getEntityById(entityId, task)
+    const deployedEntity = await getEntityById(this.components, entityId)
```

Comment:
shouldn't this be a method from the component instead of sending the component as a parameter? Something like `this.components.database.getEntityById(entityId)`?

Reply:
That would follow the old pattern where the "`repository`" component implemented a method for each query. Currently we're following other pattern in all the queries inside `database-queries`.

Reply:
can we send only the necessary components as parameters then?

Reply:
This is an emerging pattern in WKC, which is in line with tolerant readers https://martinfowler.com/bliki/TolerantReader.html: the functions only specify the components required to make it work but a broader _and possible more complete_ set of components can be passed on for clarity. the snapshots fetcher uses that a lot